qemu/tests/pvpanic.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import time
+
+from virttest import error_context
+from virttest import utils_test
+
+
+def setup_test_environment(test, params, vm, session):
+    """
+    Setup environment configuration modification.
+    Operation includds disable kdumpservice, configure
+    unknown_nmi_panic for linux, or modify the register value for
+    windows, in order to trigger crash.
+
+    :param test: test object
+    :param params: parameters
+    :vm: target vm
+    :session: session created by loggin the vm
+    """
+    timeout = int(params.get("timeout", 360))
+    if params.get("os_type") == "linux":
+        # stop kdump service and enable unknown_nmi_panic
+        setup_cmds = [params.get("set_kdump_cmd"),
+                      params.get("set_panic_cmd")]
+    else:
+        # modify the register for windows
+        setup_cmds = [params.get("set_panic_cmd")]
+    for cmd in setup_cmds:
+        status, output = session.cmd_status_output(cmd, timeout)
+        if status:
+            test.error("Command '%s' failed, status: %s, output: %s" %
+                       (cmd, status, output))
+    if params.get("os_type") == "windows":
+        vm.reboot(session, timeout=timeout)
+
+
+def check_qmp_events(vm, event_name, timeout=600):
+    """
+    Check whether certain qmp event appeared in vm.
+
+    :param vm: target virtual machine
+    :param event_name: target event name, such as 'GUEST_PANICKED'
+    :param timeout: check time
+    """
+
+    end_time = time.time() + timeout
+    qmp_monitors = vm.get_monitors_by_type("qmp")
+    logging.info("Try to get qmp events %s in %s seconds!" %
+                 (event_name, timeout))
+    while time.time() < end_time:
+        for monitor in qmp_monitors:
```

Comment:
```python
if vm.monitor.protocol != "qmp":
    logging.info("Skip qmp event check since there is no qmp monitor")
    return True
while time.time() < end_time:
    if vm.monitor.get_event(event_name):
        logging.info("Receive ... ")
        vm.monitor.clear_event(event_name)
        return True
    ...
```

Reply:
The qmp event is necessary for pvpanic test case, since the pvpanic event is only raised in qmp